tests/integration/APITest.php

```diff
@@ -64,6 +64,36 @@ public function test_set_access_token() {
 	}
 
 
+	/** @see API::perform_request() */
+	public function test_do_post_parse_response_validation_retry() {
+
+		$request = new class extends API\Request {
+
+			public function __construct() {
+
+				parent::__construct( null, null );
+
+				$this->retry_codes[] = 5678;
+			}
+		};
+
+		// mock the API to use fake data instead of making a real API call and return a valid response handler
+		$api = $this->make( facebook_for_woocommerce()->get_api(), [
+			'get_parsed_response'    => \Codeception\Stub\Expected::exactly( 6, new API\Response( json_encode( [
+				'error' => [
+					'message'          => 'Retry me!',
+					'type'             => 'OAuthException',
+					'code'             => 5678,
+				]
+			] ) ) ),
+		] );
+
+		$this->expectException( Framework\SV_WC_API_Exception::class );
```

Comment:
We may want to add a comment explaining it will be retried 5 times and then throw an exception, just to make things a bit clearer.

Reply:
Added with 35be89b24d28a7bc4dfdd3c4ab28941888353b41